### ISSUE TITLE
Add screen reader compatibility,  accessibility for the blind

### ISSUE THREAD

**Issue by azurejoga** [RHeavenStudio/HeavenStudio#654]
### Is your feature request related to a problem? Please describe.

I can navigate through the menus with the sounds when I move the keys, but knowing which option I'm on is difficult.

### Describe your suggestion

My suggestion is the following: add full support for the NVDA screen reader and using the Unity UI library, more details below

### Additional context

* playing rhythm heaven studio wouldn't be the problem, since I have a good listening level, and I've always been able to play the original rhythm heaven, however, the menus, which should be read by the screen reader, are not. making us have to guess which option we are in, even at heaven studios..
* I am completely visually impaired, 0% vision, and I would love to collaborate to make this game accessible, including the level editor.
* I will leave the resources to make games made with Unity accessible.
* several games made in unity are accessible thanks to the libraries mentioned here in this issue.
* [unity plugin (ui) plugin for integrating accessibility into unity games](https://github.com/mikrima/UnityAccessibilityPlugin)
* [how tu use](http://www.metalpopgames.com/assetstore/accessibility/doc/QuickStart.html)
* [tolk, dll / library to use screen readers in apps or games on Windows](https://github.com/dkager/tolk)
* [what is a screen reader](https://www.afb.org/blindness-and-low-vision/using-technology/assistive-technology-products/screen-readers#:~:text=Screen%20readers%20are%20software%20programs,its%20applications%2C%20and%20the%20user.)
* Feel free to ask any questions to me or other blind friends who will comment here.
# Thanks to my friends for considering accessibility as a resource, and I wish you success in the project!
If you want a Portuguese translation of the program, please, I'm here to help!

**Comment by Brunoeternal** [comment]
I am completely visually impaired, too, and I played rhythm heaven megamix and finished this. Azure has a god idea, I like this.

**Comment by minenice55** [comment]
I'll check out the suggested plugins, they don't seem to support Linux so implementing this will only happen for Windows and Mac

**Comment by azurejoga** [comment]
Oh I'm sorry!
for linux you can use [speakup](https://github.com/linux-speakup/speakup)

it is quite functional since other accessibility projects use them.
We are very happy that accessibility is being considered

**Comment by matrheine** [comment]
hey! this project has so much potential to be completely accessible to blind players indeed! rhythm heaven is one of my favorite games, I've played the versionn for GBA and 3ds on the consoles, and the wii version with the use of emulators, but unfortunately we had to memorise all the menus, so it wasn't the best experience for those without sighted help, or by using an external resource like a FAQ with text instructions to get to the desired options.
if this becomes accessible it will be a dream come true!

**Comment by azurejoga** [comment]
@matrheine  I'm just as excited as you! let's go together

**Comment by blank3times** [comment]
> I am completely visually impaired, too, and I played rhythm heaven megamix and finished this. Azure has a god idea, I like this.

In my head I've always hated on Sick Beats for being visuals-dependent (a few others might be iffy too but this is the one that I think of first), so I'm curious how that one makes itself playable! ^-^

**Comment by minenice55** [comment]
> > I am completely visually impaired, too, and I played rhythm heaven megamix and finished this. Azure has a god idea, I like this.
> 
> In my head I've always hated on Sick Beats for being visuals-dependent (a few others might be iffy too but this is the one that I think of first), so I'm curious how that one makes itself playable! ^-^

sick beats in megamix does actually use distinct audio cues for different number of skipped directions though we could differentiate the cues a bit more

**Comment by azurejoga** [comment]
@minenice55  and @blank3times 
Yes, I mainly play by listening to the audio cues and how many times we should press, in which direction we should press and which key I should press so that I don't get lost playing.
Also, the final remix is ​​quite difficult, but amazingly I finished with a not-so-bad score!
Anyway, @Brunoeternal and @matrheine would be able to explain better how a blind person plays

**Comment by Brunoeternal** [comment]
Well. This game, the sick beats, when playing, indicates in what side the virus appears, according of stereo sonds,  in right or left of speakers.

**Comment by gakishonen** [comment]
Hello.
I love the Rhythm Heaven franchise. I have been playing since 2017. Even though I am blind, I have a literal blast while playing. <a href="https://www.youtube.com/watch?v=4vZJWNcasTs">This video</a> shows how much I love the game. Unfortunately for me, the video did not show up in the recording. Too bad, I guess.
I hope you all could add screen reader support. Can't wait to see what becomes of this.

**Comment by tsatria03** [comment]
Hi there. I'm a blind individual who loves any game that is rhythm related, such as beat star and other bop it stile games in the like.
I've been playing rhythm heaven feaver since last year. I've watched countless vidios and tutorials from my friends who have played long before I was introduced to it.
I hope screen reader accessibility
 get's implemented as time goes on. I'm quite looking forward to the amazing accomplishments this game will bring in the future.

**Comment by azurejoga** [comment]
Is there any update for this?

**Comment by azurejoga** [comment]
Is there any update for this?

**Comment by ThatZeoMan** [comment]
screen reader support (as well as other languages and translations) are planned for release 2.0 at the latest, though there's a possibility it could come sooner. we're trying to figure out the best way to do it

**Comment by azurejoga** [comment]
OK! this is amazing!! I'm very happy!❤
So when you release the new versions, please let us know through this issue

**Comment by jhonata192** [comment]
I am also happy that they are thinking about accessibility and this advancement in the game will allow many people with disabilities to be included in the game.

**Comment by ThatZeoMan** [comment]
of course! we'll post updates here 🙏

**Comment by ThatZeoGal** [comment]
Alright, finally getting started on this.

https://github.com/RHeavenStudio/HeavenStudio/assets/67521686/004fdf7f-e30b-411e-879d-927b38cf5f26

As I work on this, any feedback would be incredibly helpful and appreciated! Leave feedback and suggestions here, or in our Discord server, in the "feature development" forum, in the "localization & screen reader support" post. Here's an invite to the server: https://discord.gg/rheavenstudio

The more feedback I get, the better I can make it, so please don't hesitate to reach out! The more people testing, the better :)

**Comment by azurejoga** [comment]
@ThatZeoGal  Is there already a trial version for this?
It's looking amazing! I can't wait to help with localization into Portuguese

**Comment by azurejoga** [comment]
Also, don't forget to add support for NVDA Screen Reader

**Comment by ThatZeoGal** [comment]
#

> @ThatZeoGal Is there already a trial version for this? It's looking amazing! I can't wait to help with localization into Portuguese

No trial version yet, but i’ll work on it :)
The plugin i’m using natively supports NVDA!

**Comment by ThatZeoGal** [comment]
> @ThatZeoGal I'm ...not sure what I'm looking at. Is that TTS generated by the game somehow?

Yes, the TTS (in the video is Apple’s VoiceOver) is from a screen reader system for blind and visually impaired users to use the game!

**Comment by zersiax** [comment]
Makes sense. If this is Unity I'm assuming you're using UAP? That should neatly interface with crossplatform screen readers. So you should be all good there ... if you are using the very Apple-specific Unity support we saw released last year though this will likely not work on Windows yet as is :)

**Comment by ThatZeoGal** [comment]
yes, this is UAP

**Comment by azurejoga** [comment]
@ThatZeoGal  It's awesome that you're adding mac support! However, don't forget Windows, which is the main operating system for many people, including me! If you want to release a test version in the future, don't forget to add me to the test repository as a collaborator or send me the drive link or another platform for us to download to test, I'll be here!

**Comment by ThatZeoGal** [comment]
The plugin i’m using, UAP, supports windows and mac (and linux i think) along with third-party screen readers like NVDA

**Comment by azurejoga** [comment]
This is amazing, I can't wait for a playable demo or one that I can hear in action
